docs/source/data-publishing/stac.rst

```diff
@@ -1,13 +1,305 @@
+years, and is used in numerous production deployments.
 
-pygeoapi implements STAC as an geospatial file browser through the FileSystem provider,
+pygeoapi has two built-in providers to browse STAC catalogs: `FileSystem Provider`_ and
+`Hateoas Provider`_.
+
+Hateoas Provider
+================
+
+HATEOAS (Hypermedia as the Engine of Application State) is a way of implementing a REST
+application that allows the client to dynamically navigate to the appropriate resources
+by browsing hypermedia links. This type of navigation is similar to WEB navigation
+and requires a very precise data structure that must be respected to allow the HATEOAS
+Provider to behave correctly.
+
+There are three component specifications (Catalog, Collection, Item) that together make
+up the core SpatioTemporal Asset Catalog specification. An Item represents a single
+spatiotemporal asset as GeoJSON. The Catalog specification provides structural elements,
+to group Items and Collections. Collections are catalogs, that add more required metadata
+and describe a group of related Items.
+
+The full catalog structure of links down to sub-catalogs and Items, and their links back to
+their parents and roots, must be done with **relative** URL's for the HATEOAS Provider work
+correctly. The structural *rel* types include *root*, *parent*, *child*, *item*, and
+*collection*. Assets links must be **absolute** URL's. Other links can be absolute, especially
+if they describe a resource that makes less sense in the catalog, like derived_from or even
+license (it can be nice to include the license in the catalog, but some licenses live at a
+canonical online location which makes more sense to refer to directly). This enables the
+full catalog (excluding the assets) to be downloaded or copied to another location and to
+still be valid. This also implies no self link, as that link must be absolute.
+
+So, the following rules must be respected:
+
+1. Root documents (Catalogs / Collections) must be at the root of a directory tree containing the static catalog.
+
+2. Catalogs must be named catalog.json and Collections must be named collection.json.
+
+3. Sub-Catalogs or sub-Collections must be stored in subdirectories of their parent (and only 1 subdirectory deeper than a document's parent, e.g. .../sample/sub1/catalog.json).
+
+4. Limit the number of Items in a Catalog or Collection, grouping / partitioning as relevant to the dataset.
+
+5. Use structural elements (Catalog and Collection) consistently across each 'level' of your hierarchy. For example, if levels 2 and 4 of the hierarchy only contain Collections, don't add a Catalog at levels 2 and 4.
+
+6. Items must be named <*id*>.json.
+
+7. Items must be stored in subdirectories (1 level deeper) of their parent Catalog or Collection. The subdirectory must have the same name (<*id*>) as the Item without the *.json* extension. This means that each Item are contained in a unique subdirectory.
```

Comment:
I'm wondering why there is this limitation? Additionally why have limitations on limits on depth hierarchy at all between catalogs, collections, and items? In the spec it shows catalogs, collections, and items all at the same depth.

For instance, most of the time that I encounter forecast model data, all the run outputs are in the same directory. Instead of moving all the items into their own directory, it would make sense to have them along side the source data, and potentially at the same depth as a collection containing all of those items.